sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -1514,4 +1496,100 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
       }
     }
   }
+
+  private def checkDeserializationFactor(tableName: String, exists: Boolean): Unit = {
+    spark.sessionState.catalog.refreshTable(TableIdentifier(tableName))
+    val catalogTable = getCatalogTable(tableName)
+    assert(catalogTable.stats.isDefined)
+    assert(catalogTable.stats.get.deserFactor.isDefined === exists)
+  }
+
+  private def testDeserializationFactor(fileformat: String)
+    : Unit = test(s"SPARK-24914 - test deserialization factor ($fileformat)") {
+    val table = s"sizeTest"
+
+    withTable(table) {
```

Comment:
Not sure how important this is for the tests, but my memory from an escalation is that Spark calculates sizeInBytes differently depending on whether the table is partitioned or not.

Here's my recollection (for ORC, anyway):

- Not partitioned: Spark uses totalSize
- Partitioned and spark.sql.hive.convertMetastoreOrc=true: Spark prunes the partitions and sums the size of the files in the relevant partitions.
- Partitioned and spark.sql.hive.convertMetastoreOrc=false: : Spark uses totalSize